### PR TITLE
Wrap all urllib3 read ProtocolErrors as GitProtocolError

### PR DESCRIPTION
**Pull request by nanonyme**
Fixes https://github.com/jelmer/dulwich/issues/1225

### CODE REVIEW

**Comment by nanonyme** [comment]
@jelmer not sure if this is overkill. It seemed simpler to just add an exception to protocol.py but that would have leaked HTTP client implementation details there (it looked like this has been insulated into client.py).

File: dulwich/client.py [hunk 1]
```diff
@@ -2229,6 +2229,17 @@ def __repr__(self) -> str:
         return f"{type(self).__name__}({self._base_url!r}, dumb={self.dumb!r})"
 
 
+def wrap_exceptions(func):
```
Comment (jelmer) [inline dulwich/client.py:2232]: This should probably be private and include urllib3 in the name.

**Review by jelmer — CHANGES_REQUESTED** [review]
Looks good to me overall, just one nit about the function name.

**Comment by nanonyme** [comment]
Looks like CI is failing all over due to ruff

**Review by jelmer — APPROVED** [review]

**Comment by nanonyme** [comment]
Is there any plan for next release?

**Comment by jelmer** [comment]
Not currently but I can probably do one in the next couple of weeks

**Pull request merged**